### PR TITLE
Added in new ROR search to funding body

### PR DESCRIPTION
**Pull request by lamtu1**
`FEATURE:` Add in an autocomplete search function to the new `ROR` search based into the `funding_body`.

`PROGRESS:`
`Part 1: Display`
- [x] Add in the new `funding_body` view
- [x] Update the JS script for `autocomplete` w/ `funding_body`
- [x] Create a new `QA` search for ROR

`Part 2: CV`
- [x] Create the control vocab for `ROR`
- [x] Update the `fetch` method to make the `RDF::Graph & RDF::Statement` 
- [x] Be able to save onto `curation_concern`
- [x] Index the value
- [x] Display in work `edit` page

`Part 3: Test`
- [x] Make sure it holds value
- [x] Make sure it display on show page
- [x] Make sure it display in search on SA
- [x] Make sure the facet is not an URI but with `label`
- [x] Create `spec` if needed

### CODE REVIEW

**Comment by lamtu1** [comment]
fixes #2604

**Comment by lamtu1** [comment]
`IMAGE:` Screenshot of showing the progress of the new Vocab

Screenshot 2024-11-20 at 12 46 05 PM

Screenshot 2024-11-20 at 12 46 37 PM

File: app/controllers/concerns/scholars_archive/works_controller_behavior.rb [hunk 1]
```diff
@@ -74,6 +76,35 @@ def destroy_all_files
 
     private
 
+    # METHOD: Manually add controlled_vocab object to funding body
+    def store_funding
```
Comment (straleyb) [inline app/controllers/concerns/scholars_archive/works_controller_behavior.rb:80]: I like this, but maybe we meet up and think up a neat solution about how we could do this differently.

File: lib/scholars_archive/controlled_vocabularies/research_organization_registry.rb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      private
+
+      # METHOD: Double check if rdf_label & subject are the same
+      def rdf_label_uri_same?
+        rdf_label.first.to_s == rdf_subject.to_s
+      end
+
+      # METHOD: Add in a manual fetch for graph
+      # rubocop:disable Security/Open
+      def fetch_graph_manual
```
Comment (straleyb) [inline lib/scholars_archive/controlled_vocabularies/research_organization_registry.rb:57]: Maybe we can talk about this stuff too.

**Review by straleyb — APPROVED** [review]

**Pull request merged**